packages/backend/server/src/plugins/indexer/job.ts

```diff
@@ -107,4 +125,66 @@ export class IndexerJob {
     await this.queue.remove(workspaceId, 'indexer.indexWorkspace');
     await this.service.deleteWorkspace(workspaceId);
   }
+
+  @OnJob('indexer.autoIndexWorkspaces')
+  async autoIndexWorkspaces(payload: Jobs['indexer.autoIndexWorkspaces']) {
+    const startSid = payload.lastIndexedWorkspaceSid ?? 0;
+    const workspaces = await this.models.workspace.findManyBySid(startSid, 100);
+    if (workspaces.length === 0) {
+      return JOB_SIGNAL.Repeat;
+    }
+    let addedCount = 0;
+    for (const workspace of workspaces) {
+      if (workspace.indexed) {
+        continue;
+      }
+      const snapshotMeta = await this.models.doc.getSnapshot(
+        workspace.id,
+        workspace.id,
+        {
+          select: {
+            updatedAt: true,
+          },
+        }
+      );
+      // ignore 180 days not updated workspaces
+      if (
+        !snapshotMeta?.updatedAt ||
+        Date.now() - snapshotMeta.updatedAt.getTime() >
+          180 * 24 * 60 * 60 * 1000
+      ) {
+        continue;
+      }
+      await this.queue.add(
+        'indexer.indexWorkspace',
+        { workspaceId: workspace.id },
+        { jobId: workspace.id }
+      );
+      addedCount++;
+    }
+    const nextSid = workspaces[workspaces.length - 1].sid;
+    this.logger.log(
+      `Auto added ${addedCount} workspaces to queue, lastIndexedWorkspaceSid: ${startSid} -> ${nextSid}`
+    );
+
+    // update the lastIndexedWorkspaceSid in the payload and repeat the job after 30 seconds
+    payload.lastIndexedWorkspaceSid = nextSid;
+    return JOB_SIGNAL.Repeat;
+  }
+
+  @Cron(CronExpression.EVERY_30_SECONDS)
```

Comment:
cron is not required because of handler returns `REPEAT`

Reply:
This scheduled task triggers the first task and ensures that if the task is deleted, it can be automatically resumed.